File: engine/model/src/main/scala/server/DueyPackage.scala
package server

import java.sql.Timestamp
import java.util.Calendar

import client.inventory.Item

class DueyPackage(private var _packageId: Int, private var _item: Option[Item]) {
  def packageId: Int = _packageId

  def item: Option[Item] = _item

  var sender: String = ""

  var mesos: Int = 0

  var message: String = ""

  private var timestamp: Calendar = _

  def sentTimeInMilliseconds: Long = {
    val ts: Calendar = timestamp
    if (ts != null) {
      val cal: Calendar = Calendar.getInstance()
      cal.setTime(ts.getTime)
      cal.add(Calendar.MONTH, 1)
      cal.getTimeInMillis
    } else {
      0
    }
  }

  def isDeliveringTime: Boolean = {
    val ts: Calendar = timestamp
    if (ts != null) {
      ts.getTimeInMillis >= System.currentTimeMillis()
    } else {
      false
    }
  }

  def setSentTime(ts: Timestamp, quick: Boolean): Unit = {
    val cal: Calendar = Calendar.getInstance()
    cal.setTimeInMillis(ts.getTime)
    if (quick) {
      if (System.currentTimeMillis() - ts.getTime < 24 * 60 * 60 * 1000) {
        cal.add(Calendar.DATE, -1)
      }
    }
    timestamp = cal
  }
}

File: engine/engine-model/src/main/java/server/DueyPackage.java
package server;

import java.sql.Timestamp;
import java.util.Calendar;
import java.util.Optional;

import client.inventory.Item;

public class DueyPackage {
   private final Integer packageId;

   private final Optional<Item> item;

   private String sender;

   private Integer mesos;

   private String message;

   private Calendar timestamp;

   public DueyPackage(Integer packageId, Optional<Item> item) {
      this.packageId = packageId;
      this.item = item;
   }

   public DueyPackage(Integer packageId) {
      this(packageId, Optional.empty());
   }

   public Long sentTimeInMilliseconds() {
      if (timestamp == null) {
         return 0L;
      }
      Calendar cal = Calendar.getInstance();
      cal.setTime(timestamp.getTime());
      cal.add(Calendar.MONTH, 1);
      return cal.getTimeInMillis();
   }

   public Boolean isDeliveringTime() {
      if (timestamp == null) {
         return false;
      }
      return timestamp.getTimeInMillis() >= System.currentTimeMillis();
   }

   public void setSentTime(Timestamp timestamp, Boolean quick) {
      Calendar cal = Calendar.getInstance();
      cal.setTimeInMillis(timestamp.getTime());
      if (quick) {
         if (System.currentTimeMillis() - timestamp.getTime() < 24 * 60 * 60 * 1000) {
            cal.add(Calendar.DATE, -1);
         }
      }
      this.timestamp = cal;
   }

   public Integer packageId() {
      return packageId;
   }

   public Optional<Item> item() {
      return item;
   }

   public String sender() {
      return sender;
   }

   public void setSender(String sender) {
      this.sender = sender;
   }

   public Integer mesos() {
      return mesos;
   }

   public void setMesos(Integer mesos) {
      this.mesos = mesos;
   }

   public String message() {
      return message;
   }

   public void setMessage(String message) {
      this.message = message;
   }
}
